components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/ActionManagementServiceImpl.java

```diff
@@ -140,6 +143,28 @@ public Action getActionByActionId(String actionId, String tenantDomain) throws A
         return CACHE_BACKED_DAO.getActionByActionId(actionId, IdentityTenantUtil.getTenantId(tenantDomain));
     }
 
+    @Override
+    public Action updateActionEndpointAuthentication(String actionType, String actionId, AuthType authentication,
+                                                     String tenantDomain) throws ActionMgtException {
```

Comment:
Where validations for auth type properties happen ?

Reply:
Validation happens at the API layer. 
https://github.com/wso2/identity-api-server/pull/630/files#diff-db581d52b54bc509e3a2b1f73bba32f855e95643b5823a3119d5b1220a5865fbR249:~:text=%3CAuthProperty%3E-,getActionEndpointAuthProperties,-(String%20authType